mesonbuild/backend/backends.py

```diff
@@ -594,6 +614,20 @@ def get_executable_serialisation(
             exe_wrapper = None
 
         workdir = workdir or self.environment.get_build_dir()
+
+        needs_rsp_file = can_use_rsp_file and sum(len(i) for i in cmd_args) >= get_rsp_threshold()
+
+        if needs_rsp_file:
+            hasher = hashlib.sha1()
+            hasher.update(bytes(str(cmd_args), encoding='utf-8'))
+            digest = hasher.hexdigest()
+            scratch_file = f'meson_rsp_{digest}.rsp'
+            rsp_file = os.path.join(self.environment.get_scratch_dir(), scratch_file)
+            with open(rsp_file, 'w', encoding='utf-8', newline='\n') as f:
```

Comment:
Why do you need the `newline` argument here?

Reply:
On a Windows host this will use `\r\n`, whereas glib-mkenums can only handle Unix-style newlines.